### PR TITLE
feat(select): add slot selectedLabel

### PR DESCRIPTION
**Pull request by skhamvon**
ODS-877
Add new slot to customize the label of an option once selected

### CODE REVIEW

File: packages/components/select/src/components/osds-select-option/osds-select-option.tsx [hunk 1]
```diff
@@ -70,7 +70,12 @@ export class OsdsSelectOption implements OdsSelectOptionAttribute, OdsSelectOpti
    */
   @Method()
   async getLabel() {
-    return this.el.innerText;
+    const selectedLabel = this.el.querySelector('[slot="selectedLabel"]');
+    if (selectedLabel) {
+      return selectedLabel.innerHTML;
+    } else {
```
Comment (aesteves60) [inline packages/components/select/src/components/osds-select-option/osds-select-option.tsx:76]: no else need

File: packages/components/select/src/components/osds-select/osds-select.tsx [hunk 2]
```diff
@@ -327,11 +327,8 @@ export class OsdsSelect implements OdsSelectAttribute, OdsSelectEvent, OdsSelect
         <div {...{
           class: `select-trigger${this.opened ? ' opened' : ''}${this.hasError() ? ' error' : ''}`,
         }}>
-          <div class={'label'}>
-            { (!value) ?
-              <slot name={'placeholder'}>&nbsp;</slot>
-              : this.selectedOptionLabel
-            }
+          <div class={'label'} innerHTML={ (value) ? this.selectedOptionLabel : undefined } >
```
Comment (aesteves60) [inline packages/components/select/src/components/osds-select/osds-select.tsx:330]: I think we can inject javascript with the innerHTML

Comment (aesteves60) [inline packages/components/select/src/components/osds-select/osds-select.tsx:330]:
```suggestion
          <div class={'label'} innerHTML={ value ? this.selectedOptionLabel : undefined } >
```

File: packages/components/select/src/components/osds-select/osds-select.tsx [hunk 3]
```diff
@@ -327,11 +327,8 @@ export class OsdsSelect implements OdsSelectAttribute, OdsSelectEvent, OdsSelect
...
+            { (value) ? undefined : <slot name={'placeholder'}>&nbsp;</slot> }
```
Comment (aesteves60) [inline packages/components/select/src/components/osds-select/osds-select.tsx:331]:
```suggestion
            { value ? undefined : <slot name={'placeholder'}>&nbsp;</slot> }
```

**Review by aesteves60 — CHANGES_REQUESTED** [review]

**Review by dpellier — APPROVED** [review]

**Review by aesteves60 — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Pull request merged**